app/serializers/search_result_user_serializer.rb

```diff
@@ -0,0 +1,16 @@
+class SearchResultUserSerializer < ApplicationSerializer
+  attributes :id, :username, :avatar_template, :name
+
+  def avatar_template
+    if Hash === object
+      User.avatar_template(user[:username], user[:uploaded_avatar_id])
+    else
+      user.try(:avatar_template)
+    end
+  end
+
+  def user
+    object[:user] || object
+  end
+
```

Comment:
You will still need `include_name?` method here if you build this new serializer. Sam refers to another thing on the Meta.